scapy/route6.py

```diff
@@ -105,7 +105,7 @@ def delt(self, dst, gw=None):
         l = filter(lambda x: in6_ptop(x[0]) == dst and x[1] == plen, self.routes)
         if gw:
             gw = in6_ptop(gw)
-            l = filter(lambda x: in6_ptop(x[0]) == gw, self.routes)
+            l = filter(lambda x: in6_ptop(x[2]) == gw, self.routes)
```

Comment:
Maybe you could change this to `l = [x for x in self.routes if in6_ptop(x[2]) == gw`?